src/libraries/System.Text.Json/tests/Serialization/ExceptionTests.cs

```diff
@@ -527,4 +527,100 @@ public static void UnsupportedTypeFromRoot()
+    {
+        public OpenGenericTests_StreamWithSmallBuffer() : base(SerializationWrapper.StreamSerializerWithSmallBuffer) { }
+    }
+
+    public class OpenGenericTests_Writer : OpenGenericTests
+    {
+        public OpenGenericTests_Writer() : base(SerializationWrapper.WriterSerializer) { }
+    }
+
+    public abstract class OpenGenericTests
+    {
+        private SerializationWrapper Serializer { get; }
+
+        public OpenGenericTests(SerializationWrapper serializer)
+        {
+            Serializer = serializer;
+        }
+
+        [Fact]
+        public void DeserializeOpenGeneric()
+        {
+            foreach (object[] obj in TypesWithOpenGenerics_ToDeserialize())
+            {
+                Type type = (Type)obj[0];
+
+                InvalidOperationException ex = Assert.Throws<InvalidOperationException>(() => JsonSerializer.Deserialize("", type));
+                Assert.Contains(type.ToString(), ex.ToString());
+            }
+        }
+
+        [Fact]
+        public void SerializeOpenGeneric()
+        {
+            object obj;
+
+            foreach (object[] types in TypesWithOpenGenerics_ToSerialize())
+            {
+                Type type = (Type)types[0];
+
+                if (type.GetGenericArguments().Length == 1)
+                {
+                    obj = Activator.CreateInstance(type.MakeGenericType(typeof(int)));
+                }
+                else
+                {
+                    obj = Activator.CreateInstance(type.MakeGenericType(typeof(string), typeof(int)));
+                }
+
+                ArgumentException ae = Assert.Throws<ArgumentException>(() => Serializer.Serialize(obj, type));
+            }
+
+            Type openNullableType = typeof(Nullable<>);
+
+            obj = Activator.CreateInstance(openNullableType.MakeGenericType(typeof(int)));
+            InvalidOperationException nse = Assert.Throws<InvalidOperationException>(() => Serializer.Serialize(obj, openNullableType));
+            Assert.Contains(openNullableType.ToString(), nse.ToString());
+        }
+
+        private class Test<T> { }
+
+        private static IEnumerable<object[]> TypesWithOpenGenerics_ToDeserialize()
+        {
+            yield return new object[] { typeof(Test<>) };
+            yield return new object[] { typeof(Nullable<>) };
+            yield return new object[] { typeof(List<>) };
+            yield return new object[] { typeof(List<>).MakeGenericType(typeof(Test<>)) };
+            yield return new object[] { typeof(Test<>).MakeGenericType(typeof(List<>)) };
+            yield return new object[] { typeof(Dictionary<,>) };
+            yield return new object[] { typeof(Dictionary<,>).MakeGenericType(typeof(string), typeof(Nullable<>)) };
+            yield return new object[] { typeof(Dictionary<,>).MakeGenericType(typeof(Nullable<>), typeof(string)) };
+        }
+
+        private static IEnumerable<object[]> TypesWithOpenGenerics_ToSerialize()
```

Comment:
Why do you use a subset of types for Serialization?

Reply:
Because it is non-trivial to close the generics for those types so I can create an instance in `SerializeOpenGeneric` above. The test data provides sufficient coverage for the intent of the test.

Reply:
It would be better to re-use the same set of open generics on `TypesWithOpenGenerics_ToDeserialize` for `ex = Assert.Throws<InvalidOperationException>(() => Serializer.Serialize(null, type));`

Maybe you can move that to a separate test.